src/library/itunes/itunesmacosimporter.mm

```diff
@@ -0,0 +1,288 @@
+#include "library/itunes/itunesmacosimporter.h"
+
+#import <iTunesLibrary/iTunesLibrary.h>
+
+#include <QHash>
+#include <QMultiHash>
+#include <QSqlDatabase>
+#include <QSqlQuery>
+#include <QString>
+#include <QVariant>
+#include <atomic>
+#include <memory>
+#include <optional>
+#include <utility>
+
+#include "library/itunes/itunesimporter.h"
+#include "library/libraryfeature.h"
+#include "library/queryutil.h"
+#include "library/treeitem.h"
+#include "library/treeitemmodel.h"
+
+namespace {
+
+QString qStringFrom(NSString* nsString) {
+    return [nsString cStringUsingEncoding:NSUTF8StringEncoding];
+}
+
+class ImporterImpl {
+  public:
+    ImporterImpl(QSqlDatabase& database, std::atomic<bool>& cancelImport)
+            : m_database(database), m_cancelImport(cancelImport) {
+    }
+
+    void importPlaylists(NSArray<ITLibPlaylist*>* playlists) {
+        QSqlQuery queryInsertToPlaylists(m_database);
+        queryInsertToPlaylists.prepare(
+                "INSERT INTO itunes_playlists (id, name) VALUES (:id, :name)");
+
+        QSqlQuery queryInsertToPlaylistTracks(m_database);
+        queryInsertToPlaylistTracks.prepare(
+                "INSERT INTO itunes_playlist_tracks (playlist_id, track_id, "
+                "position) VALUES (:playlist_id, :track_id, :position)");
+
+        qDebug() << "Importing playlists via native iTunesLibrary framework";
+
+        // We prefer Objective-C-style for-in loops over C++ loops when dealing
+        // with Objective-C types (both here and in the methods below) since
+        // they use Objective-C's enumeration protocols and are guaranteed to
+        // interact well with Objective-C collections.
+
+        for (ITLibPlaylist* playlist in playlists) {
+            if (m_cancelImport) {
```

Comment:
I recommend to use the explicit load operation from std::atomic instead of the implicit conversion operator.